nexus/src/app/background/tasks/support_bundle_collector.rs

```diff
@@ -568,153 +568,173 @@ impl BundleCollection<'_> {
+            .await?;
+
+        if self.request.skip_sled_info {
+            return Ok(());
+        }
+
+        let Ok(sled_client) = nexus_networking::sled_client(
+            &self.collector.datastore,
+            &self.opctx,
+            sled.id(),
+            log,
+        )
+        .await
+        else {
+            tokio::fs::write(
+                sled_path.join("error.txt"),
+                "Could not contact sled",
+            )
+            .await?;
+            return Ok(());
+        };
+
+        // NB: As new sled-diagnostic commands are added they should
+        // be added to this array so that their output can be saved
+        // within the support bundle.
+        let mut diag_cmds = futures::stream::iter([
+            save_diag_cmd_output_or_error(
+                &sled_path,
+                "zoneadm",
+                sled_client.support_zoneadm_info(),
+            )
+            .boxed(),
+            save_diag_cmd_output_or_error(
+                &sled_path,
+                "dladm",
+                sled_client.support_dladm_info(),
+            )
+            .boxed(),
+            save_diag_cmd_output_or_error(
+                &sled_path,
+                "ipadm",
+                sled_client.support_ipadm_info(),
+            )
+            .boxed(),
+            save_diag_cmd_output_or_error(
+                &sled_path,
+                "nvmeadm",
+                sled_client.support_nvmeadm_info(),
+            )
+            .boxed(),
+            save_diag_cmd_output_or_error(
+                &sled_path,
+                "pargs",
+                sled_client.support_pargs_info(),
+            )
+            .boxed(),
+            save_diag_cmd_output_or_error(
+                &sled_path,
+                "pfiles",
+                sled_client.support_pfiles_info(),
+            )
+            .boxed(),
+            save_diag_cmd_output_or_error(
+                &sled_path,
+                "pstack",
+                sled_client.support_pstack_info(),
+            )
+            .boxed(),
+            save_diag_cmd_output_or_error(
+                &sled_path,
+                "zfs",
+                sled_client.support_zfs_info(),
+            )
+            .boxed(),
+            save_diag_cmd_output_or_error(
+                &sled_path,
+                "zpool",
+                sled_client.support_zpool_info(),
+            )
+            .boxed(),
+            save_diag_cmd_output_or_error(
+                &sled_path,
+                "health-check",
+                sled_client.support_health_check(),
+            )
+            .boxed(),
+        ])
+        // Currently we execute up to 10 commands concurrently which
+        // might be doing their own concurrent work, for example
+        // collectiong `pstack` output of every Oxide process that is
+        // found on a sled.
+        .buffer_unordered(10);
```

Comment:
again, concurrent but not parallel. whether that matters depends on how much CPU-bound work these functions do...

Reply:
Above you mentioned fs operations being spawned onto the blocking thread pool -- does that mean `futures_unordered` can preform those tasks concurrently still since the await point is waiting for the result from the blocking thread typically?  I am curious because the the log collection is a synchronous task which we use `spawn_blocking` for ourselves.